test/functional/test_bmc_peer_communication.py

```diff
@@ -0,0 +1,240 @@
+from test import fixtures
+
+old_path = os.environ.get('PATH')
+new_path = '{}/bin:{}'.format(os.environ.get('PYTHONPATH'), old_path)
+conf = {}
+ivn_file = None
+fake_node0 = None
+fake_node1 = None
+global add_content0
+global add_content1
+cmd_prefix0 = [
+    'sudo ipmitool fru print 0',
+    'sudo ipmitool lan print',
+    'sudo ipmitool sensor list',
+    'sudo ipmitool sel list'
+]
+
+cmd_prefix1 = [
+    'sudo ipmitool -t 0x1e fru print 0',
+    'sudo ipmitool -t 0x1e lan print',
+    'sudo ipmitool -t 0x1e sensor list',
+    'sudo ipmitool -t 0x1e sel list',
+    'ipmitool -t 0x1e -U admin -P admin -H 192.168.188.91 power off',
+    'ipmitool -t 0x1e -U admin -P admin -H 192.168.188.91 power on',
+    'ipmitool -t 0x1e -U admin -P admin -H 192.168.188.91 power reset'
+]
+
+add_content0 = [
+    "mc_add 0x1e 1 no-device-sdrs 1 0 0 159 0 0 dynsens",
+    "mc_enable 0x1e"
+]
+add_content1 = [
+    "mc_add 0x1c 1 no-device-sdrs 1 0 0 159 0 0 dynsens",
+    "mc_enable 0x1c"
+]
+
+try:
+    from ivn.core import Topology
+except ImportError as e:
+    path_ivn = os.path.join(os.path.abspath(os.path.dirname(__file__)), "..", "ivn")
+    sys.path.append(path_ivn)
+    from ivn.core import Topology
+
+
+def saved_config_file():
+    ivn_cfg = fixtures.IvnConfig()
+    fi = tempfile.NamedTemporaryFile(delete=False)
+    yaml.safe_dump(ivn_cfg.get_ivn_info(), fi, default_flow_style=False)
+    fi.close()
+    return fi.name
+
+
+def setup_module():
+    global ivn_file
+    os.environ['PATH'] = new_path
+    ivn_file = saved_config_file()
+
+
+def teardown_module():
+    global ivn_file
+    topo = Topology(ivn_file)
+    topo.delete()
+    os.unlink(ivn_file)
+    os.environ['PATH'] = old_path
+
+
+class test_bmc_communication(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        pass
+
+    @classmethod
+    def tearDownClass(cls):
+        global fake_node0
+        global fake_node1
+        if fake_node0:
+            test_bmc_communication._stop_node(fake_node0)
+        if fake_node1:
+            test_bmc_communication._stop_node(fake_node1)
+
+    @staticmethod
+    def _stop_node(node_obj):
+        node_obj.stop()
+        node_obj.terminate_workspace()
+
+    def _start_node(self, node_info):
+        fake_node_obj = CNode(node_info)
+        fake_node_obj.init()
+        fake_node_obj.precheck()
+        fake_node_obj.start()
+        return fake_node_obj
+
+    def node_cfg_up(self, node_name, ns_name, boot_image, node_type):
+        global add_content0
+        global add_content1
+        global emu_file_for_del0
+        global emu_file_for_del1
+        fake_node = None
+        fake_node = copy.deepcopy(FakeConfig().get_node_info())
+
+        emu_dir = os.path.join(config.infrasim_data, node_type)
+        emu_file = os.path.join(emu_dir, node_type + ".emu")
+        if ns_name == "node0ns":
+            for i in add_content0:
+                os.system("echo {} >> {}".format(i, emu_file))
```

Comment:
Suggest to use python function to complete appending.
try not use shell command,  it's expensive.